src/univariate/discrete/discreteuniform.jl

```diff
@@ -85,7 +99,10 @@ function cdf(d::DiscreteUniform, x::Int)
     end
 end
 
-quantile(d::DiscreteUniform, p::Real) = iszero(p) ? d.a : d.a - 1 + ceil(Int, p * span(d))
+function quantile(d::DiscreteUniform, p::Real) 
+    T = partype(d)
+    iszero(p) ? d.a : d.a - one(T) + ceil(T, p * span(d))
+end
```

Comment:
Revert since unrelated?

Reply:
Will adapt this when reverting the DiscreteUniform to non-parametric.